plugins/browser-plugin-youtube-tracking/src/options.ts

```diff
@@ -0,0 +1,117 @@
+  (k) => MediaEventType[k]
+);
+export const DefaultEvents: MediaEventType[] = [
+  MediaEventType.Ready,
+  MediaEventType.Pause,
+  MediaEventType.Play,
+  MediaEventType.End,
+  MediaEventType.SeekStart,
+  MediaEventType.SeekEnd,
+  MediaEventType.VolumeChange,
+  MediaEventType.QualityChange,
+  MediaEventType.PlaybackRateChange,
+  MediaEventType.PercentProgress,
+];
+
+export const LEGACY_EVENTS = {
+  play: MediaEventType.Play,
+  pause: MediaEventType.Pause,
+  seek: MediaEventType.SeekEnd,
+  volumechange: MediaEventType.VolumeChange,
+  ended: MediaEventType.End,
+  error: MediaEventType.Error,
+  percentprogress: MediaEventType.PercentProgress,
+  playbackratechange: MediaEventType.PlaybackRateChange,
+  playbackqualitychange: MediaEventType.QualityChange,
+} as const;
+
+const CaptureEventToYouTubeEvent: Partial<Record<MediaEventType | YTState, keyof YT.Events>> = {
+  [MediaEventType.PlaybackRateChange]: 'onPlaybackRateChange',
+  [MediaEventType.QualityChange]: 'onPlaybackQualityChange',
+  [MediaEventType.Error]: 'onError',
+  [YTState.UNSTARTED]: 'onStateChange',
+  [MediaEventType.End]: 'onStateChange',
+  [YTState.ENDED]: 'onStateChange',
+  [YTState.PLAYING]: 'onStateChange',
+  [MediaEventType.Play]: 'onStateChange',
+  [YTState.PAUSED]: 'onStateChange',
+  [MediaEventType.Pause]: 'onStateChange',
+  [YTState.BUFFERING]: 'onStateChange',
+  [YTState.CUED]: 'onStateChange',
+};
+
+export function trackingOptionsParser(
+  { id: video, options }: YouTubeMediaTrackingConfiguration,
+  LOG?: Logger
+): TrackingOptions {
+  if (!video) throw Error('Property `id` is undefined');
+
+  const { captureEvents = ['DefaultEvents'], boundaries, ...config } = options;
+
+  let parsedEvents: TrackingOptions['captureEvents'] = [];
+  for (const ev of captureEvents) {
+    // If an event is an EventGroup, get the events from that group
+    if (ev === 'AllEvents') {
+      parsedEvents = AllEvents;
+      break;
+    } else if (ev === 'DefaultEvents') {
+      parsedEvents.push(...DefaultEvents);
+    } else if (ev in LEGACY_EVENTS) {
+      parsedEvents.push(LEGACY_EVENTS[ev as keyof typeof LEGACY_EVENTS]);
+    } else if (AllEvents.indexOf(ev as MediaEventType) !== -1) {
+      parsedEvents.push(ev as MediaEventType);
+    } else {
+      LOG?.error(`'${ev}' is not a supported event`);
+    }
+  }
+
+  // Check if any events in `conf.captureEvents` require a listener on a `YTPlayerEvent`
+  // for them to fire
+  const youtubeEvents: (keyof YT.Events)[] = [];
+  for (const ev of parsedEvents) {
+    if (CaptureEventToYouTubeEvent.hasOwnProperty(ev) && CaptureEventToYouTubeEvent[ev]) {
+      youtubeEvents.push(CaptureEventToYouTubeEvent[ev]!);
+    }
+  }
+
+  let progress: TrackingOptions['boundaries'] = undefined;
+  // If "percentprogress" is in `captureEvents`, add the `progress` object
+  // to the tracking options with the user-defined boundaries, or the default
+  if (parsedEvents.indexOf(MediaEventType.PercentProgress) !== -1) {
```

Comment:
This involves bumping our `tsconfig` `target` from `ES5` (or updating `lib` to include [`es2016`](https://github.com/microsoft/TypeScript/blob/main/src/lib/es2016.array.include.d.ts)) or higher since the default ES5 lib doesn't have `includes()` [for `Array`](https://github.com/microsoft/TypeScript/blob/ca18009b8bd5cea6a7fbcba3d7e4dd3c4633bc92/src/lib/es5.d.ts#L1304).

So a good idea, but a broad enough config change that it should probably be a dedicated PR I think?